loops/swift/code.swift

```diff
@@ -0,0 +1,12 @@
+let u = Int(CommandLine.arguments[1])!  // Get input number from command line
```

Comment:
Please avoid top-level code in Swift, that doesn't benefit from all the optimisations. In this case, it'll be really noticeable. Ideally write

```swift
@main
struct Main {
    static func main() {
        let u = Int(CommandLine.arguments[1])!  // Get input number from command line
        let r = Int.random(in: 0..<10000)  // Get random number 0 <= r < 10k
        var a: [Int] = Array(repeating: 0, count: 10000)  // Array of 10k elements initialized to 0

        for i in 0..<10000 {  // 10k outer loop iterations
            for j in 0..<100000 {  // 100k inner loop iterations
                a[i] += j % u  // Simple sum
                // [if you wanted to disable overflow check here, you can optimise this with] a[i] = a[i] &+ (j % u)  // Simple sum
            }
            a[i] += r
            // [if you wanted to disable overflow check here, you can optimise this with] a[i] = a[i] &+ r  // Add random value to each element
        }

        print(a[r])  // Print single element from array
    }
}
```

Compilation: `swiftc -O -parse-as-library -o swift/executable swift/code.swift`

Reply:
TIL wasn't aware of this, this seems to yield a slightly faster result for me:
```swift
@main
struct Main {
    static func main() {
        let u = Int(CommandLine.arguments[1])!
        let r = Int.random(in: 0..<10000)

        withUnsafeTemporaryAllocation(of: Int.self, capacity: 10000) { a in
            a.initialize(repeating: 0)
            
            for i in 0..<10000 {
                for j in 0..<100000 {
                    a[i] += j % u
                }
                
                a[i] += r
            }
            
            print(a[r])
        }
    }
}
```